website/versioned_docs/version-2.0.0/how-to-create-inventory-plugin.md

```diff
@@ -0,0 +1,82 @@
+---
+id: version-2.0.0-how-to-create-inventory-plugin
+original_id: how-to-create-inventory-plugin
+title: How To: Create an Inventory Data Provider Plugin
+---
+
+Before you create a custom inventory data provider plugin, make sure that you need one.
+- If your inventory system outputs periodic files but doesn't allow you to read inventory programmatically in real time, you may want to use the "Simple Inventory" plugin and only update the values. Refer to option 2 [here](./core-plugins-simple-inventory#how-to-sync-inventory-quantities-from-an-external-system).
+- If you use a well known third-party inventory system, there may be a community plugin already available for it. If not, consider making yours available as open source so that the community can help you maintain it.
+
+Once you're sure you need to create a custom inventory plugin, use the standard steps to create the plugin boilerplate. Beyond that, an inventory plugin needs to do the following:
+- Register an `inventoryForProductConfigurations` type function that returns current inventory in stock and other related data for a list of product configurations.
+- Optionally provide a way of editing the inventory in stock in the operator UI, by registering client components
+- Optionally track order status changes to keep track of "reserved" inventory, which is inventory that is still technically in stock but should not be considered available to sell.
+
+## Register an `inventoryForProductConfigurations` function
```

Comment:
In the docs above, you state you must add both a `inventoryForProductConfigurations` and `inventoryForProductConfiguration` function, but here you only provide docs for adding a `inventoryForProductConfigurations` with no mention of the other. Should it at least be mentioned that you need to create the other function as well?

Reply:
You actually do not need to create both. The core `inventory` plugin has both, but the singular version simply calls the multiple function. It's just for convenience to make it a bit less code when you only need one. I'll explain that here.